src/main/java/com/google/opensesame/projects/ProjectDatastore.java

```diff
@@ -44,4 +58,56 @@ public static ProjectEntity getByRepositoryId(String repositoryId) {
 
     return ProjectEntity.fromProperties(datastoreEntity.getProperties());
   }
+
+  /**
+   * Gets a project entity by its repository ID. If it doesn't exist, creates a new ProjectEntity
+   * and returns that.
+   *
+   * @param repositoryId The GitHub repository ID.
+   * @return Returns an existing ProjectEntity or a new ProjectEntity with the specified repository
+   *     ID.
+   */
+  public static ProjectEntity getByRepositoryIdOrNew(String repositoryId) {
+    ProjectEntity projectEntity = getByRepositoryId(repositoryId);
+    if (projectEntity == null) {
+      projectEntity =
+          new ProjectEntity(repositoryId, new ArrayList<String>(), new ArrayList<String>());
+    }
+
+    return projectEntity;
+  }
+
+  /**
+   * Get a list of project entities by query and fetch options.
+   *
+   * @param projectsQuery The datastore query.
+   * @param fetchOptions The fetch options for the query.
+   * @return Returns a list of project entities from the query.
+   */
+  public static List<ProjectEntity> getByQuery(Query projectsQuery, FetchOptions fetchOptions) {
+    if (!projectsQuery.getKind().equals("Project")) {
+      throw new IllegalArgumentException("Query must be made to kind 'Project'.");
+    }
+
+    PreparedQuery queryResults = datastore.prepare(projectsQuery);
+
+    List<ProjectEntity> projectEntities = new ArrayList<ProjectEntity>();
+    for (Entity datastoreEntity : queryResults.asIterable(fetchOptions)) {
+      ProjectEntity newProjectEntity =
+          ProjectEntity.fromProperties(datastoreEntity.getProperties());
+      projectEntities.add(newProjectEntity);
+    }
+
+    return projectEntities;
+  }
+
+  /**
+   * Get a list of project entities by query.
+   *
+   * @param projectsQuery The datastore query.
+   * @return Returns a list of project entities from the query.
+   */
+  public static List<ProjectEntity> getByQuery(Query projectsQuery) {
```

Comment:
nit: `getProjectsByQuery` would be more descriptive.

Reply:
I agree. Changing now!